eval.go

```diff
@@ -354,7 +354,9 @@ nextFileMatch:
 		}
 
 		if opts.ChunkMatches {
-			fileMatch.ChunkMatches = cp.fillChunkMatches(matchTreeScorer, finalCands, opts.NumContextLines, fileMatch.Language, opts.DebugScore)
+			var docScore float64
+			fileMatch.ChunkMatches, docScore = cp.fillChunkMatches(matchTreeScorer, finalCands, opts.NumContextLines, fileMatch.Language, opts.DebugScore)
+			fileMatch.addScore("doc-score", docScore, -1, opts.DebugScore)
```

Comment:
Shouldn't this replace the "atom" and "fragment" scores instead of coming in as another term? It's supposed to unify them conceptually (as it takes into account both the number of atom matches, and the strength of each match).

Reply:
Right so this is one of the confusing things about how we calculate score at moment, and possibly something I messed up in this PR (maybe I missed the second part). But essentially scoring is diveded into two parts:
- scoring candidateMatch (ala chunkmatch)
- scoring document. The max chunkmatch score is an input to this.

Looking att his PR I think I forgot to take that into account properly.